### PR TITLE
Feature cart page 

### LINKED ISSUE #14: add cart page

### PR DESCRIPTION
**Pull request by Bhakti199**
Here I added cart page. #14  On clicking `add to cart` button the item is added in cart. In cart you can increase item and vice versa. Bill card is also provided with all calculations.

Screenshot (1328)

### CODE REVIEW

File: src/Components/CartCards/BillCardForCart.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import { useFilter } from "../../Context/index";
+import { calculatePrice, discountPrice } from "./CardUtils";
+import { AiOutlineMinus } from "react-icons/ai";
+export const BillCardForCart = () => {
+  const { state } = useFilter();
+  const { cartList } = state;
+  let price = cartList.length > 0 ? calculatePrice(cartList) + 199 : 0;
+  let discountObj = cartList.length > 0 ? discountPrice(cartList) : 0;
```
Comment (tushartiwari7) [inline src/Components/CartCards/BillCardForCart.jsx]: Suggestion: You may use [nullish operator `??`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) here.

File: src/Context/FilterContext/CartUtils.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  if (itemToDecrement.count === 1) {
+    return removeItemFromCart(itemToDecrement, cartList);
+  } else {
+    return cartList.map((product) =>
+      product.cartItem._id === itemToDecrement.cartItem._id
+        ? { ...itemToDecrement, count: itemToDecrement.count - 1 }
+        : product
+    );
+
+    // if(product.cartItem._id === itemToDecrement.cartItem._id)
```
Comment (Manju-sree) [inline src/Context/FilterContext/CartUtils.js]: removed unwanted code

File: src/Components/Navbar/Navbar.jsx [hunk 3]
```diff
@@ -21,6 +25,7 @@ export const Navbar = () => {
         </Link>
         <Link
           to="/product-listing"
+          npm
```
Comment (tushartiwari7) [inline src/Components/Navbar/Navbar.jsx]: What does this npm do? seen this first time, so just curious.

File: src/Context/FilterContext/CartUtils.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+export const addItemToCart = (itemToAdd, cartList) => {
+  // console.log(cartList);
```
Comment (Manju-sree) [inline src/Context/FilterContext/CartUtils.js]: remove console.log when pushing code

File: src/Components/Navbar/Navbar.jsx [hunk 5]
```diff
@@ -8,7 +8,11 @@ import { FiLogOut } from "react-icons/fi";
 import { Link } from "react-router-dom";
 import { useFilter } from "../../Context/index";
 export const Navbar = () => {
-  const { dispatch } = useFilter();
+  const { dispatch, state } = useFilter();
+  const { cartList } = state;
+  const sum = (acc, curr) => (acc = acc + curr.count);
+  const cartCount = cartList.length > 0 ? cartList.reduce(sum, 0) : 0;
```
Comment (JantuDeb) [inline src/Components/Navbar/Navbar.jsx:14]: This is really good how you calculating total price

File: src/Context/FilterContext/CartUtils.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        : product
+    );
+
+    // if(product.cartItem._id === itemToDecrement.cartItem._id)
+    // return{
+    //     ...itemToDecrement,
+    //     count: itemToDecrement.count - 1
+    //   }else{
+    //     return product;
+    //   }
```
Comment (tushartiwari7) [inline src/Context/FilterContext/CartUtils.js]: remove commented code

**Review by Manju-sree — APPROVED** [review]
please check calculation part in cart page and please remove code which is not required before pushing changes to git

**Review by tushartiwari7 — APPROVED** [review]
Suggested some changes please have a look.

File: src/Context/FilterContext/CartUtils.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    }
+  }
+};
+
+export const removeItemFromCart = (itemToRemove, cartList) => {
+  console.log({ itemToRemove, cartList });
+  return cartList.filter(
+    (product) => product.cartItem._id !== itemToRemove.cartItem._id
+  );
+};
```
Comment (JantuDeb) [inline src/Context/FilterContext/CartUtils.js]:
```
export const removeItemFromCart = (itemToRemove, cartList) =>cartList.filter((product) => product.cartItem._id !== itemToRemove.cartItem._id);
```
remove console.log from final build

File: src/Context/FilterContext/CartUtils.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+export const incrementCartItem = (itemToIncrement, cartList) => {
+  console.log({ itemToIncrement, cartList });
+  return cartList.map((product) =>
+    product.cartItem._id === itemToIncrement.cartItem._id
+      ? { ...itemToIncrement, count: itemToIncrement.count + 1 }
+      : product
+  );
+};
```
Comment (JantuDeb) [inline src/Context/FilterContext/CartUtils.js]:
```
export const incrementCartItem = (itemToIncrement, cartList) => cartList.map((product) =>
    product.cartItem._id === itemToIncrement.cartItem._id
 ? { ...itemToIncrement, count: itemToIncrement.count + 1 } : product  );
```

File: src/Context/FilterContext/FilterReducer.js [hunk 9]
```diff
@@ -77,9 +84,32 @@ export const filterManagement = (state, action) => {
...
+    case "INCREMENT_CART_ITEM":
+      return {
+        ...state,
+        cartList: incrementCartItem(action.payload, state.cartList),
+      };
+    case "DECREMENT_CART_ITEM":
+      return {
+        ...state,
+        cartList: decrementCartItem(action.payload, state.cartList),
+      };
```
Comment (JantuDeb) [inline src/Context/FilterContext/FilterReducer.js:110]: This is really good that you have separated all the functions.

**Review by Manju-sree — CHANGES_REQUESTED** [review]
please check once calculation is not happening properly in cart

File: src/Context/FilterContext/CartUtils.js [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+export const decrementCartItem = (itemToDecrement, cartList) => {
+  if (itemToDecrement.count === 1) {
+    return removeItemFromCart(itemToDecrement, cartList);
+  } else {
+    return cartList.map((product) =>
```
Comment (JantuDeb) [inline src/Context/FilterContext/CartUtils.js]: No need to write `else` you can directly `return` it.

**Review by JantuDeb — APPROVED** [review]
I have added some comments

File: src/Components/Navbar/Navbar.jsx (hunk 3 above)
Comment (Bhakti199) [reply src/Components/Navbar/Navbar.jsx]: It is by mistake

**Pull request merged**